language/diem-vm/src/diem_transaction_executor.rs

```diff
@@ -150,16 +150,16 @@ impl DiemVM {
         ))
     }
 
-    fn execute_script<R: RemoteCache>(
+    fn execute_script_or_script_function<R: RemoteCache>(
```

Comment:
Since we are doing the switch here anyway, should we just introduce another function called `execute_script_function`?

Reply:
We originally did a split of `execute_script_function` and `execute_script` but we later realized that there are too much code duplication between them, so we combined them into one function.

Reply:
I like having it as one function, but how would you feel about leaving the name as "execute_script"? I'm all in favor of clear and descriptive names but that one is getting kind of long and "script" could be interpreted to include script functions as well.

Reply:
`execute_script_alike`?

Reply:
"alike"? Sorry, I don't get it

Reply:
I guess what I was trying to say is that: executing sth like a script. But I am terrible at giving names... In any way, I think I can tolerate `execute_script_or_script_function` as it is an internal function and is only used in the same file.

Reply:
Ah I see. The only reason why I feel disalike about this api is that we expect the caller to already matched on `TransactionPayload` and make sure the `Modules` variant is unreachable. To me being able to get rid of unneeded matches like this seems desirable. But I do hear that having a uniform api seems nice so I'm a bit torn over there..